scripts/unpack

```diff
@@ -98,7 +105,53 @@ if [ -d "$SOURCES/$1" -o -d "$PKG_DIR/sources" ]; then
     cp -PRf $PKG_DIR/sources/* $BUILD/${PKG_NAME}-${PKG_VERSION}
   fi
 
-  if [ -z "$PKG_URL" ]; then
+  fi
+
+  if [ -z "$PKG_URL" -a -n "$PKG_GIT_URL" ]; then
+    # init GIT_ORIG_DIR
+    GIT_ORIG_DIR=""
+    GIT_PREV_LOCATION=$(pwd)
+    # cycle through folders that match $PKG_NAME in $BUILD until we find the right one
+    for f in $BUILD/$1-*/ ; do
+      if [ -d $f ]; then
+        if [ -f $f/.libreelec-unpack ]; then
+          source $f/.libreelec-unpack
+          if [ "$PKG_NAME" = "$STAMP_PKG_NAME" ]; then
+            cd $f
+            BUILD_GIT_URL=$(git remote get-url origin)
+            if [ "$BUILD_GIT_URL" = "$PKG_GIT_URL" ]; then
+              # git cloned repository found
+              GIT_ORIG_DIR=$f
+              break
+            fi
+          fi
+        fi
+      fi
+    done
+    cd $GIT_PREV_LOCATION
+    if [ -d "$GIT_ORIG_DIR" -a -n "$GIT_ORIG_DIR" ]; then
+      if [ ! "$GIT_ORIG_DIR" = "$BUILD/$PKG_NAME-$PKG_VERSION" ]; then
+        mv $GIT_ORIG_DIR $BUILD/$PKG_NAME-$PKG_VERSION
+      fi
+      cd $BUILD/$PKG_NAME-$PKG_VERSION
+      # clean repository from any local changes
+      git clean -fdx
+      git checkout -- .
+      # switch to master branch and pull
+      git checkout master
+      git pull
+    else
+      # no valid folder found => clean
+      $SCRIPTS/clean $1
+      # clone the repository
+      git clone $PKG_GIT_URL $BUILD/$PKG_NAME-$PKG_VERSION/
```

Comment:
--single-branch --depth=1 can be used?

Reply:
--single-branch should be OK, but --depth=1 sure not - you need the whole history to checkout the specific commit defined in PKG_VERSION - if you would clone only with 1 commit history (last commit to the primary branch), you would probably fail checkout an earlier commit.

Reply:
And what about  `git submodule update --init --recursive --depth 1`?